lib/chef/provider/windows_task.rb

```diff
@@ -202,8 +204,9 @@ def task_need_update?
             @current_resource.frequency != new_resource.frequency ||
             @current_resource.idle_time != new_resource.idle_time ||
             @current_resource.random_delay != new_resource.random_delay ||
-            @current_resource.execution_time_limit != new_resource.execution_time_limit ||
-            !new_resource.start_day.nil? || !new_resource.start_time.nil?
+            !new_resource.execution_time_limit.include?(@current_resource.execution_time_limit) ||
+            (!new_resource.start_day.nil? && start_day_updated?) ||
```

Comment:
`new_resource.start_day` instead of `!new_resource.start_day.nil?` seems like a better pattern here.

Reply:
(and above and below)